From: Philip Withnall <pwithnall@gnome.org>
Date: Tue, 10 Dec 2024 16:59:33 +0000
Subject: gdbusconnection: Document existing refcount semantics of closures
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

As per the previous few commits, explicitly document the established
reference counting semantics of the method call closure for
`g_dbus_connection_register_object_with_closures()`.

This isn’t ideal, but
`g_dbus_connection_register_object_with_closures()` has had these
semantics for 10 years now, and it’s a bit late to change them to
something neater.

Signed-off-by: Philip Withnall <pwithnall@gnome.org>
Bug: https://gitlab.gnome.org/GNOME/glib/-/issues/3559
Forwarded: https://gitlab.gnome.org/GNOME/glib/-/merge_requests/4427
---
 gio/gdbusconnection.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/gio/gdbusconnection.c b/gio/gdbusconnection.c
index a4b57cf..8e8e16a 100644
--- a/gio/gdbusconnection.c
+++ b/gio/gdbusconnection.c
@@ -6089,6 +6089,11 @@ register_with_closures_on_set_property (GDBusConnection *connection,
  * Version of g_dbus_connection_register_object() using closures instead of a
  * #GDBusInterfaceVTable for easier binding in other languages.
  *
+ * Note that the reference counting semantics of the function wrapped by
+ * @method_call_closure are the same as those of
+ * [callback@Gio.DBusInterfaceMethodCallFunc]: ownership of a reference to the
+ * [class@Gio.DBusMethodInvocation] is transferred to the function.
+ *
  * Returns: 0 if @error is set, otherwise a registration ID (never 0)
  * that can be used with g_dbus_connection_unregister_object() .
  *
